<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Methods</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>




<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DISTAL Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="getting-started.html">Getting started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="02-fig2.html">Figure 2</a>
    </li>
    <li>
      <a href="05-fig3.html">Figure 3</a>
    </li>
    <li>
      <a href="03-fig4.html">Figure 4</a>
    </li>
    <li>
      <a href="04-suppfig2.html">Supplementary figure 2</a>
    </li>
    <li>
      <a href="06-suppfig4.html">Supplementary figure 4</a>
    </li>
    <li>
      <a href="05-fig3.html#Supplementary_figure_10">Supplementary figure 10</a>
    </li>
    <li>
      <a href="05-fig3.html#Supplementary_figure_11">Supplementary figure 11</a>
    </li>
    <li>
      <a href="05-fig3.html#Supplementary_figure_12">Supplementary figure 12</a>
    </li>
    <li>
      <a href="07-suppfig14.html">Supplementary figure 14</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Reports
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="cellranger_web_summary.html">Cell Ranger</a>
    </li>
  </ul>
</li>
<li>
  <a href="90-methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/Chuang1118/FB5P-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Methods</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-12-18
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>proj_distal/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongnoversioncontrol"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Repository version:</strong> no version control </a>
</p>
</div>
<div id="strongRepositoryversionstrongnoversioncontrol" class="panel-collapse collapse">
<div class="panel-body">
<p>Tracking code development and connecting the code version to the results is critical for reproducibility. To start using Git, open the Terminal and type <code>git init</code> in your project directory.</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
This project is not being versioned with Git. To obtain the full reproducibility benefits of using workflowr, please see <code>?wflow_start</code>.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code># Presentation
library(&quot;glue&quot;)
library(&quot;knitr&quot;)

# JSON
library(&quot;jsonlite&quot;)

# Tidyverse
library(&quot;tidyverse&quot;)</code></pre>
<pre class="r"><code>dir.create(here::here(&quot;output&quot;, DOCNAME), showWarnings = FALSE)

write_bib(c(&quot;base&quot;, &quot;Seurat&quot;, &quot;cowplot&quot;, &quot;ggplot2&quot;, &quot;ggforce&quot;,
            &quot;gridExtra&quot;, &quot;tidyverse&quot;, &quot;dplyr&quot;, &quot;tidyr&quot;, &quot;purrr&quot;,
             &quot;workflowr&quot;, &quot;knitr&quot;, &quot;rmarkdown&quot;),
          file = here::here(&quot;output&quot;, DOCNAME, &quot;packages.bib&quot;))</code></pre>
<pre class="r"><code>docs &lt;- list.dirs(here::here(&quot;output&quot;), full.names = FALSE)[-1]

params &lt;- map(docs, function(doc) {
    out &lt;- tryCatch(
        {
            table &lt;- suppressWarnings(
                read_json(here::here(glue(&quot;output/{doc}/parameters.json&quot;)),
                          simplifyVector = TRUE)
            )
            p_list &lt;- table$Value %&gt;% setNames(table$Parameter)
        },
        error = function(e) {
            message(glue(&quot;{doc} parameters file not found&quot;))

            return(list())
        }
    )    
    
    return(out)
})

names(params) &lt;- str_remove(docs, &quot;[0-9A-Z]+\\-&quot;)
names(params) &lt;- names(params) %&gt;%
    str_replace(&quot;preprocessing&quot;, &quot;pre&quot;) %&gt;%
    str_replace(&quot;quality-control&quot;, &quot;qc&quot;) %&gt;%
    str_replace(&quot;clustering&quot;, &quot;clust&quot;) %&gt;%
    str_replace(&quot;marker-genes&quot;, &quot;markers&quot;)

versions &lt;- list(
    cellranger     = &quot;3.0.1&quot;,
    cellranger_ref = &quot;3.0.0&quot;,
    cowplot        = packageVersion(&quot;cowplot&quot;),
    dplyr          = packageVersion(&quot;dplyr&quot;),
    DropletUtils   = packageVersion(&quot;DropletUtils&quot;),
    edgeR          = packageVersion(&quot;edgeR&quot;),
    ensembl        = &quot;93&quot;,
    ggforce        = packageVersion(&quot;ggforce&quot;),
    ggplot2        = packageVersion(&quot;ggplot2&quot;),
    gridExtra      = packageVersion(&quot;gridExtra&quot;),
    knitr          = packageVersion(&quot;knitr&quot;),
    pandoc         = rmarkdown::pandoc_version(),
    purrr          = packageVersion(&quot;purrr&quot;),
    R              = str_extract(R.version.string, &quot;[0-9\\.]+&quot;),
    rmarkdown      = packageVersion(&quot;rmarkdown&quot;),
    scater         = packageVersion(&quot;scater&quot;),
    Seurat         = packageVersion(&quot;Seurat&quot;),
    tidyr          = packageVersion(&quot;tidyr&quot;),
    tidyverse      = packageVersion(&quot;tidyverse&quot;),
    viridis        = packageVersion(&quot;viridis&quot;),
    workflowr      = packageVersion(&quot;workflowr&quot;)
)</code></pre>
<div id="pre-processing" class="section level1">
<h1>Pre-processing</h1>
<p>The FB5P-seq pipeline (v1) <span class="citation">(Attaf N 2020)</span> was used to perform sample demultiplexing, barcode processing and single-cell gene counting. Briefly, single cells of interest are sorted in 96-well plates by FACS, routinely using a 10-color staining strategy to identify and enrich specific subsets of B or T cells while recording all parameters through index sorting. Reads containing sequence information were aligned using the GRCh38 reference genome and ENSEMBL gene annotation (v93). PCR duplicates were removed by selecting unique combinations of cell barcodes, unique molecular identifiers and gene ids with the final results being a gene expression matrix that was used for further analysis. The R statistical programming language (v4.1.2) <span class="citation">(R Core Team 2021)</span> was used for further analysis. Count data was read into R and used to construct a Seurat object (v4.1.0) <span class="citation">(Hoffman 2022)</span>.</p>
<!-- Droplet selection -->
<!-- ----------------- -->
<!-- The droplet selection method in this version of Cell Ranger identified -->
<!--  cell containing droplets. This is more than the -->
<!-- traditional Cell Ranger approach based on a simple threshold on total counts -->
<!-- which identified  cells. The EmptyDrops method  -->
<!-- [@Lun2019-wp] available in the DropletUtils package (v1.14.2)  -->
<!-- was also applied. Droplets with less than  total -->
<!-- counts were used to construct the ambient RNA profile and an FDR threshold of -->
<!-- less than  was used to select   -->
<!-- droplets. The droplets selected by either the Cell Ranger v3 method or -->
<!-- EmptyDrops were kept giving a dataset with  cells.  cells were -->
<!-- assigned cell cycle scores using the cyclone [@Scialdone2015-rp] function in the -->
<!-- scran package (v) [@Lun2016-mq]. -->
</div>
<div id="quality-control" class="section level1">
<h1>Quality control</h1>
<p>The ggplot2 package (v3.4.4) <span class="citation">(Wickham 2010)</span> was used to produce a series of diagnostic quality control plots and select thresholds for filtering. Cells were removed from the dataset if they had total features expressed less than 400 features, Accuracy Pearson ERCC less than 70%, the percentage of mitochondrial gene more than 12% and Percent ERCC more than 25%.</p>
</div>
<div id="clustering" class="section level1">
<h1>Clustering</h1>
<div id="gene-selection" class="section level2">
<h2>Gene selection</h2>
<p>Two methods were used to select genes to use for clustering. The default selection method in the Seurat package (v4.1.0) <span class="citation">(Satija et al. 2015)</span> uses thresholds on mean expression (between and ) and dispersion (greater than ), which selected<br />
genes. The M3Drop (v) <span class="citation">(Andrews and Hemberg 2018)</span> method was also used to test for genes with significantly more zero counts than expected and genes with an adjusted p-value less than were selected. Following comparisions between these two methods the genes identified by M3Drop were used for clustering.</p>
</div>
<div id="graph-based-clustering" class="section level2">
<h2>Graph-based clustering</h2>
<p>Seurat was used to perform graph-based clustering. PCA was performed using the selected genes and the first principal components were used to construct a shared nearest neighbour graph using the overlap between the nearest neighbours of each cell. Louvain modularity optimisation <span class="citation">(Blondel et al. 2008)</span> was used to partition this graph with a resolution parameter between  and -. Clustering tree visualisations <span class="citation">(Zappia and Oshlack 2018)</span> were produced using the clustree package (v) showing the expression of previously identified marker genes. By inspecting these trees a resolution of which chosen which produced clusters. To compare these cluster to those that had been previously published the Jaccard index was calculated between pairs of clusters from each analysis and visualised as a heatmap.</p>
</div>
</div>
<div id="marker-genes" class="section level1">
<h1>Marker genes</h1>
<p>Marker genes for each cluster were identified using edgeR (v3.36.0) <span class="citation">(Robinson, McCarthy, and Smyth 2010; McCarthy, Chen, and Smyth 2012)</span>. Additional filtering was performed to remove genes with a <span class="math inline">\(\log_{10}\)</span> maximum average counts in any cluster less than . The edgeR negative binomial model was then fitted to the dataset using a design matrix that included the detection rate (scaled proportion of genes expressed in each cell). The quasi-likelihood F-test was used to test cells in one cluster against all other cells. Genes were considered significant markers for a cluster if they had an FDR less than and a log foldchange greater than . Identities were assigned to each cluster by comparing the detected genes to previously published lists of cell type markers.</p>
</div>
<div id="other-packages" class="section level1">
<h1>Other packages</h1>
<p>Visualisations and figures were primarily created using the ggplot2 (v3.4.4) <span class="citation">(Wickham 2010)</span> and cowplot (v1.1.1) <span class="citation">(Wilke 2020)</span> packages using the viridis colour palette (v0.6.2) for continuous data. UpSet plots <span class="citation">(Lex et al. 2014)</span> were produced using the UpSetR package (v) <span class="citation">(<strong>R-UpSetR?</strong>)</span> with help from the gridExtra package (v2.3) <span class="citation">(Auguie 2017)</span> and SinaPlots <span class="citation">(Sidiropoulos et al. 2018)</span> using the ggforce package (v0.3.3) <span class="citation">(Pedersen 2021)</span>. Data manipulation was performed using other packages in the tidyverse (v1.3.1) <span class="citation">(Wickham 2021)</span> particularly dplyr (v1.0.8) <span class="citation">(Wickham et al. 2022)</span>, tidyr (v1.0.8) <span class="citation">(Wickham and Girlich 2022)</span> and purrr (v0.3.4) <span class="citation">(Henry and Wickham 2020)</span>.The analysis project was managed using the workflowr (v1.7.1) <span class="citation">(Blischak, Carbonetto, and Stephens 2023)</span> package which was also used to produce the publicly available website displaying the analysis code, results and output. Reproducible reports were produced using knitr (v1.37) <span class="citation">(Xie 2014, 2016, 2021)</span> and R Markdown (v2.11) <span class="citation">(Xie, Allaire, and Grolemund 2018; Allaire et al. 2021)</span> and converted to HTML using Pandoc (v2.14.0.3).</p>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-R-rmarkdown" class="csl-entry">
Allaire, JJ, Yihui Xie, Jonathan McPherson, Javier Luraschi, Kevin Ushey, Aron Atkins, Hadley Wickham, Joe Cheng, Winston Chang, and Richard Iannone. 2021. <em>Rmarkdown: Dynamic Documents for r</em>.
</div>
<div id="ref-Andrews2018-xa" class="csl-entry">
Andrews, T S, and Martin Hemberg. 2018. <span>“<span class="nocase">M3Drop: Dropout-based feature selection for scRNASeq</span>.”</span> <em>Bioinformatics</em>, December. <a href="https://doi.org/10.1093/bioinformatics/bty1044">https://doi.org/10.1093/bioinformatics/bty1044</a>.
</div>
<div id="ref-Attaf2020-fi" class="csl-entry">
Attaf N, Dong C, Cervera-Marzal I. 2020. <span>“<span class="nocase">FB5P-seq: FACS-Based 5-Prime End Single-Cell RNA-seq for Integrative Analysis of Transcriptome and Antigen Receptor Repertoire in B and T Cells</span>.”</span> <em>Frontiers in Immunology</em> 11 (1): 216. <a href="https://doi.org/10.3389/fimmu.2020.00216">https://doi.org/10.3389/fimmu.2020.00216</a>.
</div>
<div id="ref-R-gridExtra" class="csl-entry">
Auguie, Baptiste. 2017. <em>gridExtra: Miscellaneous Functions for "Grid" Graphics</em>.
</div>
<div id="ref-R-workflowr" class="csl-entry">
Blischak, John, Peter Carbonetto, and Matthew Stephens. 2023. <em>Workflowr: A Framework for Reproducible and Collaborative Data Science</em>. <a href="https://github.com/workflowr/workflowr">https://github.com/workflowr/workflowr</a>.
</div>
<div id="ref-Blondel2008-ym" class="csl-entry">
Blondel, Vincent D, Jean-Loup Guillaume, Renaud Lambiotte, and Etienne Lefebvre. 2008. <span>“<span class="nocase">Fast unfolding of communities in large networks</span>.”</span> <em>Journal of Statistical Mechanics</em> 2008 (10): P10008. <a href="https://doi.org/10.1088/1742-5468/2008/10/P10008">https://doi.org/10.1088/1742-5468/2008/10/P10008</a>.
</div>
<div id="ref-R-purrr" class="csl-entry">
Henry, Lionel, and Hadley Wickham. 2020. <em>Purrr: Functional Programming Tools</em>.
</div>
<div id="ref-R-Seurat" class="csl-entry">
Hoffman, Paul. 2022. <em>Seurat: Tools for Single Cell Genomics</em>.
</div>
<div id="ref-Lex2014-hy" class="csl-entry">
Lex, Alexander, Nils Gehlenborg, Hendrik Strobelt, Romain Vuillemot, and Hanspeter Pfister. 2014. <span>“<span class="nocase">UpSet: Visualization of Intersecting Sets</span>.”</span> <em>IEEE Transactions on Visualization and Computer Graphics</em> 20 (12): 1983–92. <a href="https://doi.org/10.1109/TVCG.2014.2346248">https://doi.org/10.1109/TVCG.2014.2346248</a>.
</div>
<div id="ref-McCarthy2012-gc" class="csl-entry">
McCarthy, Davis J, Yunshun Chen, and Gordon K Smyth. 2012. <span>“<span class="nocase">Differential expression analysis of multifactor RNA-Seq experiments with respect to biological variation</span>.”</span> <em>Nucleic Acids Research</em> 40 (10): 4288–97. <a href="https://doi.org/10.1093/nar/gks042">https://doi.org/10.1093/nar/gks042</a>.
</div>
<div id="ref-R-ggforce" class="csl-entry">
Pedersen, Thomas Lin. 2021. <em>Ggforce: Accelerating Ggplot2</em>.
</div>
<div id="ref-R-base" class="csl-entry">
R Core Team. 2021. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-Robinson2010-pt" class="csl-entry">
Robinson, Mark D, Davis J McCarthy, and Gordon K Smyth. 2010. <span>“<span class="nocase">edgeR: a Bioconductor package for differential expression analysis of digital gene expression data</span>.”</span> <em>Bioinformatics</em> 26 (1): 139–40. <a href="https://doi.org/10.1093/bioinformatics/btp616">https://doi.org/10.1093/bioinformatics/btp616</a>.
</div>
<div id="ref-Satija2015-or" class="csl-entry">
Satija, Rahul, Jeffrey A Farrell, David Gennert, Alexander F Schier, and Aviv Regev. 2015. <span>“<span class="nocase">Spatial reconstruction of single-cell gene expression data</span>.”</span> <em>Nature Biotechnology</em> 33 (5): 495–502. <a href="https://doi.org/10.1038/nbt.3192">https://doi.org/10.1038/nbt.3192</a>.
</div>
<div id="ref-Sidiropoulos2018-ll" class="csl-entry">
Sidiropoulos, Nikos, Sina Hadi Sohi, Thomas Lin Pedersen, Bo Torben Porse, Ole Winther, Nicolas Rapin, and Frederik Otzen Bagger. 2018. <span>“<span class="nocase">SinaPlot: An Enhanced Chart for Simple and Truthful Representation of Single Observations Over Multiple Classes</span>.”</span> <em>Journal of Computational and Graphical Statistics: A Joint Publication of American Statistical Association, Institute of Mathematical Statistics, Interface Foundation of North America</em> 27 (3): 673–76. <a href="https://doi.org/10.1080/10618600.2017.1366914">https://doi.org/10.1080/10618600.2017.1366914</a>.
</div>
<div id="ref-Wickham2010-zq" class="csl-entry">
Wickham, Hadley. 2010. <em><span class="nocase">ggplot2: Elegant Graphics for Data Analysis</span></em>. Springer New York.
</div>
<div id="ref-R-tidyverse" class="csl-entry">
———. 2021. <em>Tidyverse: Easily Install and Load the Tidyverse</em>.
</div>
<div id="ref-R-dplyr" class="csl-entry">
Wickham, Hadley, Romain François, Lionel Henry, and Kirill Müller. 2022. <em>Dplyr: A Grammar of Data Manipulation</em>.
</div>
<div id="ref-R-tidyr" class="csl-entry">
Wickham, Hadley, and Maximilian Girlich. 2022. <em>Tidyr: Tidy Messy Data</em>.
</div>
<div id="ref-R-cowplot" class="csl-entry">
Wilke, Claus O. 2020. <em>Cowplot: Streamlined Plot Theme and Plot Annotations for Ggplot2</em>. <a href="https://wilkelab.org/cowplot/">https://wilkelab.org/cowplot/</a>.
</div>
<div id="ref-Xie2014-ha" class="csl-entry">
Xie, Yihui. 2014. <span>“<span class="nocase">knitr: A Comprehensive Tool for Reproducible Research in R</span>.”</span> In <em><span>Implementing Reproducible Research</span></em>, edited by Victoria Stodden, Friedrich Leisch, and Roger D Peng. CRC Press. <a href="https://market.android.com/details?id=book-JcmSAwAAQBAJ">https://market.android.com/details?id=book-JcmSAwAAQBAJ</a>.
</div>
<div id="ref-Xie2016-ct" class="csl-entry">
———. 2016. <em><span class="nocase">Dynamic Documents with R and knitr</span></em>. CRC Press. <a href="https://doi.org/10.1201/b15166">https://doi.org/10.1201/b15166</a>.
</div>
<div id="ref-R-knitr" class="csl-entry">
———. 2021. <em>Knitr: A General-Purpose Package for Dynamic Report Generation in r</em>. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
<div id="ref-Xie2018-tw" class="csl-entry">
Xie, Yihui, J J Allaire, and Garrett Grolemund. 2018. <em><span>R Markdown: The Definitive Guide</span></em>. CRC Press LLC. <a href="https://market.android.com/details?id=book-j9YyuAEACAAJ">https://market.android.com/details?id=book-j9YyuAEACAAJ</a>.
</div>
<div id="ref-Zappia2018-lz" class="csl-entry">
Zappia, Luke, and Alicia Oshlack. 2018. <span>“<span class="nocase">Clustering trees: a visualization for evaluating clusterings at multiple resolutions</span>.”</span> <em>GigaScience</em> 7 (7). <a href="https://doi.org/10.1093/gigascience/giy083">https://doi.org/10.1093/gigascience/giy083</a>.
</div>
</div>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<div id="output-files" class="section level2">
<h2>Output files`</h2>
<p>This table describes the output files produced by this document. Right click and <em>Save Link As…</em> to download the results.</p>
<pre class="r"><code>params &lt;- jsonlite::toJSON(params, pretty = TRUE)
readr::write_lines(params,
                   here::here(&quot;output&quot;, DOCNAME, &quot;analysis-parameters.json&quot;))

versions &lt;- purrr::map(versions, as.character)
versions &lt;- jsonlite::toJSON(versions, pretty = TRUE)
readr::write_lines(versions,
                   here::here(&quot;output&quot;, DOCNAME, &quot;package-versions.json&quot;))</code></pre>
<pre class="r"><code>dir.create(here::here(&quot;output&quot;, DOCNAME), showWarnings = FALSE)

knitr::kable(data.frame(
    File = c(
        
    ),
    Description = c(

    )
))</code></pre>
<p>|| || || ||</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>devtools::session_info()</code></pre>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.1.2 (2021-11-01)
 os       Ubuntu 20.04.3 LTS
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       Etc/UTC
 date     2024-12-18
 pandoc   2.14.0.3 @ /usr/lib/rstudio-server/bin/pandoc/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 package     * version date (UTC) lib source
 assertthat    0.2.1   2019-03-21 [1] RSPM (R 4.1.0)
 backports     1.4.1   2021-12-13 [1] RSPM (R 4.1.0)
 bit           4.0.4   2020-08-04 [1] RSPM (R 4.1.0)
 bit64         4.0.5   2020-08-30 [1] RSPM (R 4.1.0)
 brio          1.1.3   2021-11-30 [1] RSPM (R 4.1.0)
 broom         0.7.12  2022-01-28 [1] RSPM (R 4.1.0)
 bslib         0.3.1   2021-10-06 [1] RSPM (R 4.1.0)
 cachem        1.0.6   2021-08-19 [1] RSPM (R 4.1.0)
 callr         3.7.0   2021-04-20 [1] RSPM (R 4.1.0)
 cellranger    1.1.0   2016-07-27 [1] RSPM (R 4.1.0)
 cli           3.6.1   2023-03-23 [1] RSPM (R 4.1.0)
 colorspace    2.0-3   2022-02-21 [1] RSPM (R 4.1.0)
 crayon        1.5.0   2022-02-14 [1] RSPM (R 4.1.0)
 DBI           1.1.2   2021-12-20 [1] RSPM (R 4.1.0)
 dbplyr        2.1.1   2021-04-06 [1] RSPM (R 4.1.0)
 desc          1.4.1   2022-03-06 [1] RSPM (R 4.1.0)
 devtools      2.4.3   2021-11-30 [1] RSPM (R 4.1.0)
 digest        0.6.29  2021-12-01 [1] RSPM (R 4.1.0)
 dplyr       * 1.0.8   2022-02-08 [1] RSPM (R 4.1.0)
 ellipsis      0.3.2   2021-04-29 [1] RSPM (R 4.1.0)
 evaluate      0.15    2022-02-18 [1] RSPM (R 4.1.0)
 fansi         1.0.2   2022-01-14 [1] RSPM (R 4.1.0)
 fastmap       1.1.0   2021-01-25 [1] RSPM (R 4.1.0)
 forcats     * 0.5.1   2021-01-27 [1] RSPM (R 4.1.0)
 fs            1.5.2   2021-12-08 [1] RSPM (R 4.1.0)
 generics      0.1.2   2022-01-31 [1] RSPM (R 4.1.0)
 ggplot2     * 3.4.4   2023-10-12 [1] RSPM (R 4.1.0)
 git2r         0.33.0  2023-11-26 [1] RSPM (R 4.1.0)
 glue        * 1.6.2   2022-02-24 [1] RSPM (R 4.1.0)
 gtable        0.3.0   2019-03-25 [1] RSPM (R 4.1.0)
 haven         2.4.3   2021-08-04 [1] RSPM (R 4.1.0)
 here          1.0.1   2020-12-13 [1] RSPM (R 4.1.0)
 hms           1.1.1   2021-09-26 [1] RSPM (R 4.1.0)
 htmltools     0.5.2   2021-08-25 [1] RSPM (R 4.1.0)
 httpuv        1.6.5   2022-01-05 [1] RSPM (R 4.1.0)
 httr          1.4.2   2020-07-20 [1] RSPM (R 4.1.0)
 jquerylib     0.1.4   2021-04-26 [1] RSPM (R 4.1.0)
 jsonlite    * 1.8.0   2022-02-22 [1] RSPM (R 4.1.0)
 knitr       * 1.37    2021-12-16 [1] RSPM (R 4.1.0)
 later         1.3.0   2021-08-18 [1] RSPM (R 4.1.0)
 lifecycle     1.0.3   2022-10-07 [1] RSPM (R 4.1.0)
 lubridate     1.8.0   2021-10-07 [1] RSPM (R 4.1.0)
 magrittr      2.0.2   2022-01-26 [1] RSPM (R 4.1.0)
 memoise       2.0.1   2021-11-26 [1] RSPM (R 4.1.0)
 modelr        0.1.8   2020-05-19 [1] RSPM (R 4.1.0)
 munsell       0.5.0   2018-06-12 [1] RSPM (R 4.1.0)
 pillar        1.7.0   2022-02-01 [1] RSPM (R 4.1.0)
 pkgbuild      1.3.1   2021-12-20 [1] RSPM (R 4.1.0)
 pkgconfig     2.0.3   2019-09-22 [1] RSPM (R 4.1.0)
 pkgload       1.2.4   2021-11-30 [1] RSPM (R 4.1.0)
 prettyunits   1.1.1   2020-01-24 [1] RSPM (R 4.1.0)
 processx      3.5.2   2021-04-30 [1] RSPM (R 4.1.0)
 promises      1.2.0.1 2021-02-11 [1] RSPM (R 4.1.0)
 ps            1.6.0   2021-02-28 [1] RSPM (R 4.1.0)
 purrr       * 0.3.4   2020-04-17 [1] RSPM (R 4.1.0)
 R6            2.5.1   2021-08-19 [1] RSPM (R 4.1.0)
 Rcpp          1.0.8   2022-01-13 [1] RSPM (R 4.1.0)
 readr       * 2.1.1   2021-11-30 [1] RSPM (R 4.1.0)
 readxl        1.3.1   2019-03-13 [1] RSPM (R 4.1.0)
 remotes       2.4.2   2021-11-30 [1] RSPM (R 4.1.0)
 reprex        2.0.1   2021-08-05 [1] RSPM (R 4.1.0)
 rlang         1.1.1   2023-04-28 [1] RSPM (R 4.1.0)
 rmarkdown     2.11    2021-09-14 [1] RSPM (R 4.1.0)
 rprojroot     2.0.2   2020-11-15 [1] RSPM (R 4.1.0)
 rstudioapi    0.13    2020-11-12 [1] RSPM (R 4.1.0)
 rvest         1.0.2   2021-10-16 [1] RSPM (R 4.1.0)
 sass          0.4.0   2021-05-12 [1] RSPM (R 4.1.0)
 scales        1.2.1   2022-08-20 [1] RSPM (R 4.1.0)
 sessioninfo   1.2.2   2021-12-06 [1] RSPM (R 4.1.0)
 stringi       1.7.6   2021-11-29 [1] RSPM (R 4.1.0)
 stringr     * 1.4.0   2019-02-10 [1] RSPM (R 4.1.0)
 testthat      3.1.2   2022-01-20 [1] RSPM (R 4.1.0)
 tibble      * 3.1.8   2022-07-22 [1] RSPM (R 4.1.0)
 tidyr       * 1.2.0   2022-02-01 [1] RSPM (R 4.1.0)
 tidyselect    1.1.2   2022-02-21 [1] RSPM (R 4.1.0)
 tidyverse   * 1.3.1   2021-04-15 [1] RSPM (R 4.1.0)
 tzdb          0.2.0   2021-10-27 [1] RSPM (R 4.1.0)
 usethis       2.1.5   2021-12-09 [1] RSPM (R 4.1.0)
 utf8          1.2.2   2021-07-24 [1] RSPM (R 4.1.0)
 vctrs         0.6.4   2023-10-12 [1] RSPM (R 4.1.0)
 vroom         1.5.7   2021-11-30 [1] RSPM (R 4.1.0)
 withr         2.5.0   2022-03-03 [1] RSPM (R 4.1.0)
 workflowr     1.7.1   2023-08-23 [1] RSPM (R 4.1.0)
 xfun          0.30    2022-03-02 [1] RSPM (R 4.1.0)
 xml2          1.3.3   2021-11-30 [1] RSPM (R 4.1.0)
 yaml          2.3.5   2022-02-21 [1] RSPM (R 4.1.0)

 [1] /usr/local/lib/R/site-library
 [2] /usr/local/lib/R/library

──────────────────────────────────────────────────────────────────────────────</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.8     purrr_0.3.4    
 [5] readr_2.1.1     tidyr_1.2.0     tibble_3.1.8    ggplot2_3.4.4  
 [9] tidyverse_1.3.1 jsonlite_1.8.0  knitr_1.37      glue_1.6.2     

loaded via a namespace (and not attached):
 [1] httr_1.4.2        sass_0.4.0        pkgload_1.2.4     bit64_4.0.5      
 [5] vroom_1.5.7       here_1.0.1        modelr_0.1.8      brio_1.1.3       
 [9] bslib_0.3.1       assertthat_0.2.1  cellranger_1.1.0  remotes_2.4.2    
[13] yaml_2.3.5        sessioninfo_1.2.2 pillar_1.7.0      backports_1.4.1  
[17] digest_0.6.29     promises_1.2.0.1  rvest_1.0.2       colorspace_2.0-3 
[21] htmltools_0.5.2   httpuv_1.6.5      pkgconfig_2.0.3   devtools_2.4.3   
[25] broom_0.7.12      haven_2.4.3       scales_1.2.1      processx_3.5.2   
[29] later_1.3.0       tzdb_0.2.0        git2r_0.33.0      generics_0.1.2   
[33] usethis_2.1.5     ellipsis_0.3.2    cachem_1.0.6      withr_2.5.0      
[37] cli_3.6.1         magrittr_2.0.2    crayon_1.5.0      readxl_1.3.1     
[41] memoise_2.0.1     evaluate_0.15     ps_1.6.0          fs_1.5.2         
[45] fansi_1.0.2       xml2_1.3.3        pkgbuild_1.3.1    tools_4.1.2      
[49] prettyunits_1.1.1 hms_1.1.1         lifecycle_1.0.3   munsell_0.5.0    
[53] reprex_2.0.1      callr_3.7.0       compiler_4.1.2    jquerylib_0.1.4  
[57] rlang_1.1.1       grid_4.1.2        rstudioapi_0.13   rmarkdown_2.11   
[61] testthat_3.1.2    gtable_0.3.0      DBI_1.1.2         R6_2.5.1         
[65] lubridate_1.8.0   fastmap_1.1.0     bit_4.0.4         utf8_1.2.2       
[69] workflowr_1.7.1   rprojroot_2.0.2   desc_1.4.1        stringi_1.7.6    
[73] parallel_4.1.2    Rcpp_1.0.8        vctrs_0.6.4       dbplyr_2.1.1     
[77] tidyselect_1.1.2  xfun_0.30        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
