---
title: "Methods"
bibliography:
    - "`r here::here('data/references/references.bib')`"
    - "`r here::here('output/90-methods/packages.bib')`"
---

```{r knitr, include = FALSE}
DOCNAME = "90-methods"
NOW <- Sys.time()

# Time chunks during knitting
knitr::knit_hooks$set(timeit = function(before) {
    if (before) {
        print(paste("Start:", Sys.time()))
        NOW <<- Sys.time()
    } else {
        print(paste("Stop:", Sys.time()))
        print(Sys.time() - NOW)
    }
})

knitr::opts_chunk$set(
    autodep        = TRUE,
    cache          = FALSE,
    cache.path     = paste0("cache/", DOCNAME, "/"),
    cache.comments = FALSE,
    echo           = TRUE,
    error          = FALSE,
    fig.align      = "center",
    fig.width      = 10,
    fig.height     = 8,
    message        = FALSE,
    warning        = FALSE,
    timeit         = TRUE
)
```

```{r libaries, cache = FALSE}
# Presentation
library("glue")
library("knitr")

# JSON
library("jsonlite")

# Tidyverse
library("tidyverse")
```

```{r source, cache = FALSE}

```

```{r pkg-bib}
dir.create(here::here("output", DOCNAME), showWarnings = FALSE)

write_bib(c("base", "Seurat", "cowplot", "ggplot2", "ggforce",
            "gridExtra", "tidyverse", "dplyr", "tidyr", "purrr",
             "workflowr", "knitr", "rmarkdown"),
          file = here::here("output", DOCNAME, "packages.bib"))
```

```{r load}
docs <- list.dirs(here::here("output"), full.names = FALSE)[-1]

params <- map(docs, function(doc) {
    out <- tryCatch(
        {
            table <- suppressWarnings(
                read_json(here::here(glue("output/{doc}/parameters.json")),
                          simplifyVector = TRUE)
            )
            p_list <- table$Value %>% setNames(table$Parameter)
        },
        error = function(e) {
            message(glue("{doc} parameters file not found"))

            return(list())
        }
    )    
    
    return(out)
})

names(params) <- str_remove(docs, "[0-9A-Z]+\\-")
names(params) <- names(params) %>%
    str_replace("preprocessing", "pre") %>%
    str_replace("quality-control", "qc") %>%
    str_replace("clustering", "clust") %>%
    str_replace("marker-genes", "markers")

versions <- list(
    cellranger     = "3.0.1",
    cellranger_ref = "3.0.0",
    cowplot        = packageVersion("cowplot"),
    dplyr          = packageVersion("dplyr"),
    DropletUtils   = packageVersion("DropletUtils"),
    edgeR          = packageVersion("edgeR"),
    ensembl        = "93",
    ggforce        = packageVersion("ggforce"),
    ggplot2        = packageVersion("ggplot2"),
    gridExtra      = packageVersion("gridExtra"),
    knitr          = packageVersion("knitr"),
    pandoc         = rmarkdown::pandoc_version(),
    purrr          = packageVersion("purrr"),
    R              = str_extract(R.version.string, "[0-9\\.]+"),
    rmarkdown      = packageVersion("rmarkdown"),
    scater         = packageVersion("scater"),
    Seurat         = packageVersion("Seurat"),
    tidyr          = packageVersion("tidyr"),
    tidyverse      = packageVersion("tidyverse"),
    viridis        = packageVersion("viridis"),
    workflowr      = packageVersion("workflowr")
)
```

Pre-processing
=============

The FB5P-seq pipeline (v1) [@Attaf2020-fi] was used to 
perform sample demultiplexing, barcode processing and single-cell gene counting. 
Briefly, single cells of interest are sorted in 96-well plates by FACS, routinely 
using a 10-color staining strategy to identify and enrich specific subsets of B or 
T cells while recording all parameters through index sorting. Reads containing sequence 
information were aligned using the GRCh38 reference genome and ENSEMBL gene 
annotation (v`r versions$ensembl`). PCR duplicates were removed by selecting 
unique combinations of cell barcodes, unique molecular  identifiers and gene ids 
with the final results being a gene expression matrix that was used for 
further analysis. The R statistical programming language (v`r versions$R`) [@R-base] 
was used for further analysis. Count data  was read  into R and used to construct a Seurat object 
(v`r versions$Seurat`) [@R-Seurat].

<!-- Droplet selection -->
<!-- ----------------- -->

<!-- The droplet selection method in this version of Cell Ranger identified -->
<!-- `r params$pre$n_cellranger` cell containing droplets. This is more than the -->
<!-- traditional Cell Ranger approach based on a simple threshold on total counts -->
<!-- which identified `r params$pre$n_default` cells. The EmptyDrops method  -->
<!-- [@Lun2019-wp] available in the DropletUtils package (v`r versions$DropletUtils`)  -->
<!-- was also applied. Droplets with less than `r params$pre$empty_thresh` total -->
<!-- counts were used to construct the ambient RNA profile and an FDR threshold of -->
<!-- less than `r params$pre$emp_fdr` was used to select `r params$pre$n_empdrops`  -->
<!-- droplets. The droplets selected by either the Cell Ranger v3 method or -->
<!-- EmptyDrops were kept giving a dataset with `r params$pre$n_cells` cells.  cells were -->
<!-- assigned cell cycle scores using the cyclone [@Scialdone2015-rp] function in the -->
<!-- scran package (v`r versions$scran`) [@Lun2016-mq]. -->


Quality control
===============

The ggplot2 package (v`r versions$ggplot2`) [@Wickham2010-zq] was used to produce
a series of diagnostic quality control plots and select thresholds for filtering. 
Cells were removed from the dataset if they had total features expressed less than 400 features, 
Accuracy Pearson ERCC less than 70%, the percentage of mitochondrial gene more than 12% and 
Percent ERCC more than 25%.

Clustering
==========

Gene selection
--------------

Two methods were used to select genes to use for clustering. The default
selection method in the Seurat package (v`r versions$Seurat`) [@Satija2015-or]
uses thresholds on mean expression (between `r params$clust$seurat_thresh[1]`
and `r params$clust$seurat_thresh[2]`) and dispersion (greater than
`r params$clust$seurat_thresh[3]`), which selected `r params$clust$sel_seurat` 
genes. The M3Drop (v`r versions$M3Drop`) [@Andrews2018-xa] method was also used
to test for genes with significantly more zero counts than expected and
`r params$clust$sel_m3drop` genes with an adjusted p-value less than
`r params$clust$m3drop_thresh` were selected. Following comparisions between
these two methods the genes identified by M3Drop were used for clustering.

Graph-based clustering
----------------------

Seurat was used to perform graph-based clustering. PCA was performed using the
selected genes and the first `r params$clust$n_pcs` principal components were
used to construct a shared nearest neighbour graph using the overlap between
the `r params$clust$knn` nearest neighbours of each cell. Louvain modularity
optimisation [@Blondel2008-ym] was used to partition this graph with a
resolution parameter between `r min(params$clust$resolutions)` and
`r max(params$clust$resolutions)`. Clustering tree visualisations
[@Zappia2018-lz] were produced using the clustree package
(v`r versions$clustree`) showing the expression of previously identified
marker genes. By inspecting these trees a resolution of `r params$clust$res`
which chosen which produced `r params$clust$n_clusts` clusters. To compare these
cluster to those that had been previously published the Jaccard index was
calculated between pairs of clusters from each analysis and visualised as a
heatmap.

Marker genes
============

Marker genes for each cluster were identified using edgeR (v`r versions$edgeR`)
[@Robinson2010-pt; @McCarthy2012-gc]. Additional filtering was performed to
remove genes with a $\log_{10}$ maximum average counts in any cluster less than
`r params$markers$maxmean_thresh`. The edgeR negative binomial model was then 
fitted to the dataset using a design matrix that included the detection rate
(scaled proportion of genes expressed in each cell). The quasi-likelihood F-test
was used to test cells in one cluster against all other cells. Genes were
considered significant markers for a cluster if they had an FDR less than
`r params$markers$fdr_thresh` and a log foldchange greater than
`r params$markers$logFC_thresh`. Identities were assigned to each cluster by
comparing the detected genes to previously published lists of cell type markers.


Other packages
==============

Visualisations and figures were primarily created using the
ggplot2 (v`r versions$ggplot2`) [@Wickham2010-zq] and cowplot
(v`r versions$cowplot`) [@R-cowplot] packages using the viridis colour palette
(v`r versions$viridis`) for continuous data. UpSet plots [@Lex2014-hy] were 
produced using the UpSetR package (v`r versions$UpSetR`) [@R-UpSetR] with help
from the gridExtra package (v`r versions$gridExtra`) [@R-gridExtra] and
SinaPlots [@Sidiropoulos2018-ll] using the ggforce package
(v`r versions$ggforce`) [@R-ggforce]. Data manipulation was performed using
other packages in the tidyverse (v`r versions$tidyverse`) [@R-tidyverse]
particularly dplyr (v`r versions$dplyr`) [@R-dplyr], tidyr (v`r versions$dplyr`) 
[@R-tidyr] and purrr (v`r versions$purrr`) [@R-purrr].The analysis project was 
managed using the workflowr (v`r versions$workflowr`) [@R-workflowr] package 
which was also used to produce the publicly available website displaying the 
analysis code, results and output. Reproducible reports were produced using knitr (v`r versions$knitr`) 
[@Xie2014-ha; @Xie2016-ct; @R-knitr] and R Markdown (v`r versions$rmarkdown`) 
[@Xie2018-tw; @R-rmarkdown] and converted to HTML using Pandoc
(v`r versions$pandoc`). 

References
==========

<div id="refs"></div>

Summary
=======

Output files`
------------

This table describes the output files produced by this document. Right click
and _Save Link As..._ to download the results.

```{r save}
params <- jsonlite::toJSON(params, pretty = TRUE)
readr::write_lines(params,
                   here::here("output", DOCNAME, "analysis-parameters.json"))

versions <- purrr::map(versions, as.character)
versions <- jsonlite::toJSON(versions, pretty = TRUE)
readr::write_lines(versions,
                   here::here("output", DOCNAME, "package-versions.json"))
```

```{r output}
dir.create(here::here("output", DOCNAME), showWarnings = FALSE)

knitr::kable(data.frame(
    File = c(
        
    ),
    Description = c(

    )
))
```

Session information
-------------------

```{r session-info, cache = FALSE}
devtools::session_info()
```
